src/nbl/asset/utils/CGeometryCreator.cpp

```diff
@@ -398,258 +399,615 @@ core::smart_refctd_ptr<ICPUPolygonGeometry> CGeometryCreator::createSphere(float
 		u2 = core::normalize(core::cross(u2, v0ToApex)) * angleWeight;
 
-		baseVertices[i].normal = quantNormalCache->quantize<EF_A2B10G10R10_SNORM_PACK32>(core::normalize(u1 + u2));
-		apexVertices[i].normal = quantNormalCache->quantize<EF_A2B10G10R10_SNORM_PACK32>(core::normalize(u1));
+
+		const auto baseNormal = quantNormalCache->quantize<NormalCacheFormat>(core::normalize(u1 + u2));
+		memcpy(normals + i, &baseNormal, sizeof(baseNormal));
+
+		const auto apexNormal = quantNormalCache->quantize<NormalCacheFormat>(core::normalize(u1));
+		memcpy(normals + apexVertexBase_i + i, &apexNormal, sizeof(apexNormal));
 	}
 
-	auto idxBuf = asset::ICPUBuffer::create({ 3u * tesselation * sizeof(uint16_t) });
-	uint16_t* indices = (uint16_t*)idxBuf->getPointer();
+	CPolygonGeometryManipulator::recomputeContentHashes(retval.get());
+	return retval;
+}
 
-	const uint32_t firstIndexOfBaseVertices = 0;
-	const uint32_t firstIndexOfApexVertices = tesselation;
-	for (uint32_t i = 0; i < tesselation; i++)
+core::smart_refctd_ptr<ICPUPolygonGeometry> CGeometryCreator::createArrow(
+	const uint32_t tesselationCylinder,
+	const uint32_t tesselationCone,
+	const float height,
+	const float cylinderHeight,
+	const float width0,
+	const float width1,
+	const video::SColor vtxColor0,
+	const video::SColor vtxColor1
+) const
+{
+  assert(height > cylinderHeight);
+
+	using position_t = hlsl::float32_t3;
+	using normal_t = hlsl::vector<uint8_t, 4>;
+	using uv_t = hlsl::vector<uint8_t, 2>;
+	using color_t = hlsl::vector<uint8_t, 4>;
+
+  auto cylinder = createCylinder(width0, cylinderHeight, tesselationCylinder, vtxColor0);
+  auto cone = createCone(width1, height-cylinderHeight, tesselationCone, vtxColor1, vtxColor1);
+
+	auto cylinderPositions = reinterpret_cast<position_t*>(cylinder->getPositionView().src.buffer->getPointer());
+	auto conePositions = reinterpret_cast<position_t*>(cone->getPositionView().src.buffer->getPointer());
+
+	const auto cylinderNormals = reinterpret_cast<normal_t*>(cylinder->getNormalView().src.buffer->getPointer());
+	const auto coneNormals = reinterpret_cast<normal_t*>(cone->getNormalView().src.buffer->getPointer());
+
+	const auto cylinderUvs = reinterpret_cast<uv_t*>(cylinder->getAuxAttributeViews()->front().src.buffer->getPointer());
+	const auto coneUvs = reinterpret_cast<uv_t*>(cone->getAuxAttributeViews()->front().src.buffer->getPointer());
+
+	const auto cylinderIndices = cylinder->getIndexView().src.buffer->getPointer();
+	const auto coneIndices = cone->getIndexView().src.buffer->getPointer();
+
+	const auto cylinderVertexCount = cylinder->getPositionView().getElementCount();
+	const auto coneVertexCount = cone->getPositionView().getElementCount();
+	const auto newArrowVertexCount = cylinderVertexCount + coneVertexCount;
+
+	const auto cylinderIndexCount = cylinder->getVertexReferenceCount();
+	const auto coneIndexCount = cone->getVertexReferenceCount();
+	const auto newArrowIndexCount = cylinderIndexCount + coneIndexCount;
```

Comment:
Actually. make this function return a geometry collection!

Its a perfect use case! (untouched geometries, just transforms applied)

You can skip rendering the arrow in ex09